src/app/features/reporting/reporting/reporting.component.ts

```diff
@@ -54,36 +65,66 @@ export class ReportingComponent {
     fromDate: Date,
     toDate: Date,
   ) {
-    this.loading = true;
+    this.isError = false;
+    this.errorDetails = null;
+    this.isLoading = true;
     this.data = [];
 
-    // Wait for change detection
-    await new Promise((res) => setTimeout(res));
+    if (this.reportCalculation) {
+      // trigger re-calculation
+      await firstValueFrom(
+        this.sqlReportService
+          .createReportCalculation(selectedReport.getId(), fromDate, toDate)
+          .pipe(
+            map((value) => value.id),
+            switchMap((id) =>
+              lastValueFrom(this.sqlReportService.waitForReportData(id)),
+            ),
+            switchMap((value) =>
+              // todo handle FINISHED_ERROR case
+              this.sqlReportService.fetchReportCalculationData(value.id),
+            ),
```

Comment:
feels like this would better go into the service as a utility method? Considering we have different report approaches (except sql also) handled here in parallel, keeping the sql/api logic out of the component?

Reply:
could we move the whole re-calculation logic into the service and even hide it behind the sqlService.query method, maybe?

Reply:
done